src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+the underlying native Android and iOS UI libraries, attempting to smooth out the
+inconsistencies of each platform representation. App code is often written in an
+interpreted language like JavaScript, which must in turn interact with the
+Java-based Android or Objective-C-based iOS system libraries to display UI. All
+this adds overhead that can be significant, particularly where there is a lot of
+interaction between the UI and the app logic.
+
+By contrast, Flutter minimizes those abstractions, bypassing the system UI
+widget libraries in favor of its own widget set. The Dart code that paints
+Flutter’s visuals is compiled into native code, which uses Skia for rendering.
+Flutter also embeds its own copy of Skia as part of the engine, allowing the
+developer to upgrade their app to stay updated with the latest performance
+improvements even if the phone hasn’t been updated with a new Android version.
+The same is true for Flutter on other native platforms, such as iOS, Windows, or
+macOS.
+
+### From user input to the GPU
+
+The overriding principle Flutter applies to its rendering pipeline is that
+**simple is fast**. Flutter has a straightforward pipeline for how data flows to
+the system, as shown in the following sequencing diagram:
+
+![Render pipeline sequencing
+diagram](../../images/arch-overview/render-pipeline.svg){:width="100%"}
+
+Let’s take a look at some of these phases in greater detail.
+
+### Build: from Widget to Element
+
+Consider this simple code fragment that demonstrates a simple widget hierarchy:
+
+```dart
+Container(
+  color: Colors.blue,
+  child: Row(
+    children: [
+      Image.network('https://www.example.com/1.png'),
+      Text('A'),
+    ],
+  ),
+);
+```
+
+When Flutter needs to render this fragment, it calls the `build()` method, which
+returns a subtree of widgets that render UI based on the current app state.
+During this process, the `build()` method can introduce new widgets, as
+necessary, based on its state. As a simple example, in the preceding code
+fragment, `Container` has `color` and `child` properties. From looking at the
+[source
+code](https://github.com/flutter/flutter/blob/f7a6a7906be96d2288f5d63a5a54c515a6e987fe/packages/flutter/lib/src/widgets/container.dart#L433)
+for `Container`, you can see that if the color is not null, it inserts a
+`ColoredBox` representing the color:
+
+```dart
+if (color != null)
+  current = ColoredBox(color: color, child: current);
+```
+
+Correspondingly, the `Image` and `Text` widgets might insert child widgets such
+as `RawImage` and `RichText` during the build process. The eventual widget
+hierarchy may therefore be deeper than what the code represents, as in this
+case<sup><a href="#a2">2</a></sup>:
+
+![Render pipeline sequencing
+diagram](../../images/arch-overview/widgets.svg){:width="100%"}
+
+This explains why, when you examine the tree through a debug tool such as the
+[DartDev widget
```

Comment:
We don't use that name elsewhere. 
DartDev widget inspector => Flutter inspector
OR
=> Flutter inspector, part of the Dart DevTools.